postgres/organisation_service.go

```diff
@@ -51,6 +51,31 @@ func (o *OrganisationService) AddOrganisation(
 	return convertOrganisation(organisation)
 }
 
+// Calls UpdateOrganisation query
+func (o *OrganisationService) UpdateOrganisation(
+	ctx context.Context,
+	organisationID core.OrganisationID,
+	name string,
+) error {
+	return o.UpdateOrganisationTx(ctx, o.db, organisationID, name)
+}
+
+// Calls UpdateOrganisation query
+func (o *OrganisationService) UpdateOrganisationTx(
+	ctx context.Context,
+	dbtx sqlc.DBTX,
+	organisationID core.OrganisationID,
+	name string,
+) error {
+	queries := sqlc.New(dbtx)
+	intID := int32(organisationID)
+	err := queries.UpdateOrganisation(ctx, intID, name)
+	if err != nil {
+		return ConvertPgError(err)
+	}
+	return nil
+}
+
```

Comment:
Als je die hier zelf niet nodig hebt als transactie moet je dat ook niet toevoegen, dat is enkel extra code die we moeten onderhouden. Zo'n dingen voeg je best enkel toe op het moment dat je dat echt nodig hebt. Voor een update waar je enkel de naam bij kan aanpassen lijkt mij ook niet realistisch dat je dat ooit in een transactie zou moeten kunnen doen

Reply:
Ja goed punt, ik gebruik het inderdaad nog niet. Al lijkt het me wel nuttig om in de toekomst te hebben, om bijvoorbeeld ook de facturatiegegevens van een organisatie aan te passen in 1 transactie etc. Desalniettemin heb ik de functie terug verwijderd

Reply:
Ik denk niet dat iemand snel tegelijk beide gegevens gaat willen aanpassen 😛 en zelfs dan nog is het perfect ok om dat in 2x te doen. Bij aanmaken is dat niet het geval, als 1 van de 2 queries faalt kom je in een ongeldige toestand terecht en dat wil je voorkomen door die transaction te gebruiken. Strikt genomen kan je het daar ook zonder transaction doen maar dat wordt snel heel ingewikkeld (als de 2e query faalt moet je het eerste object dan opnieuw verwijderen maar de query die dat doet kan ook falen en dan zit je helemaal in een ambetante situatie 😛)